petab/v1/problem.py

```diff
@@ -1227,6 +1208,16 @@ class ProblemConfig(BaseModel):
         description="The base path to resolve relative paths.",
         exclude=True,
     )
-    format_version: VersionNumber = 1
+    format_version: str | int = 1
     parameter_file: str | AnyUrl | None = None
     problems: list[SubProblem] = []
+
+    def to_yaml(self, filename: str | Path):
+        """Write the configuration to a YAML file.
+
+        filename: Destination file name. The parent directory will be created
+            if necessary.
```

Comment:
Not in `__all__` so I guess it doesn't matter for docs, but usually this style in this repo
```suggestion
        Parameters:
            filename:
                Destination file name. The parent directory will be
                created if necessary.
```
.. I think. In the files touched by this PR alone, I see
```
Parameters
----------

Arguments:

Parameters:
```

Reply:
Yeah, it would make sense to start a style guide.... 

I'd prefer `:param ...: `

Reply:
Agreed, but no preference for the specific style

Reply:
Started a style guide in https://github.com/PEtab-dev/libpetab-python/pull/354